### ISSUE TITLE
Mesos unique node constraints?

### ISSUE THREAD

**Issue by gdhgdhgdh** [dhilipkumars/MrRedis#38]
Hi,

When I issue the following command, I hoped that there would be a high change for the tasks to be spread across multiple Mesos agents:

```
curl -X POST http://mrredis.mesos:5656/v1/CREATE/redistyk/256/1/3
```

I was really surprised to see three out of the four total `redistyk` tasks running on the same agent (even with huge spare capacity across the cluster!)

Can I request the ability to specify the framework equivalent of  `"constraints": [["hostname", "UNIQUE"]]` in a Marathon job?

**Comment by dhilipkumars** [comment]
In-order to create instances really quickly we accept one offer for the master -> wait for it to be created -> accept offer and create slaves.

This means that if the second offer is big enough to hold all the three slaves then we simply choose that.

Sure this request can be done, would you like an option to spread it across different nodes? 
like

```
curl -X POST http://mrredis.mesos:5656/v1/CREATE/redistyk/256/1/3 -d  
{
   "spread":3
}
```

if we do not get 3 nodes to spreads then what happens? 
1) we could update the status from 'CREATING' to 'INSUFFICIENT RESOURCE' after retrying for certain amount of time.
2) we could keep trying forever until we get an offer like that.

**Comment by gdhgdhgdh** [comment]
Hmm, if the `spread` param has been specified, it seems reasonable to also pass through `retries` and `interval` and if not specified, `retries` would be infinite?

**Comment by dhilipkumars** [comment]
Addressed by 3399b67440497a39ffc3951133dc0617efb6e4a9 and d62665d508da863fcdc7105b00cd08b6fcc32622
